includes/class-wcs-cart-renewal.php

```diff
@@ -1654,6 +1657,33 @@ public function set_renewal_order_cart_hash_on_block_checkout( $has_status, $ord
 		return $has_status;
 	}
 
+	/**
+	 * Restores the order awaiting payment session args if the cart contains a subscription-related order.
+	 *
+	 * It's possible the that order_awaiting_payment and store_api_draft_order session args are not set if those session args are lost due
+	 * to session destruction.
+	 *
+	 * This function checks the cart that is being loaded from the session and if the cart contains a subscription-related order and if the
+	 * current user has permission to pay for it. If so, it restores the order awaiting payment session args.
+	 *
+	 * @param WC_Cart $cart The cart object.
+	 */
+	public function restore_order_awaiting_payment( $cart ) {
+
```

Comment:
Another non-essential, but we could add some defensive coding here. Possibly:

```suggestion
		if ( ! is_a( $cart, WC_Cart::class ) ) {
			return;
		}
```

Or else perhaps:

```suggestion
		if ( ! method_exists( $cart, 'get_cart' ) ) {
			return;
		}
```

Thinking that we've seen a few cases where callbacks expect to receive a certain type, but get something else. That's less likely here, because the hook is an action (and so other callbacks in the chain cannot modify the _thing)_ ... but also isn't impossible (because a third party can trigger the action and pass whatever it wants, and because a future iteration of WooCommerce itself could *conceivably* start passing other types such as `WP_Error`).